src/common.c

```diff
@@ -1688,7 +1688,7 @@ static int ofi_pollfds_waitall(struct ofi_pollfds *pfds,
 		if (!found && timeout > 0)
 			timeout -= (int) (ofi_gettime_ms() - start);
 
-	} while (timeout > 0 && !found);
+	} while (timeout > 0 || !found);
```

Comment:
This effective disables the timeout parameter. The proper way is to use different variables to track the timeout passed in and the time left (if timeout > 0).

Reply:
I clearly didn't think this one through very thoroughly. I think it can still work with just the one timeout variable if the while condition is `(timeout != 0 && !found)`. Just requires different code at line 1689. I'll push another commit and see what you think.

Reply:
The updated code looks fine.